docs/guides/gradle.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Gradle configuration
+headline: Documentation
+bodyclass: docs
+layout: docs
+---
+
+# Configuring a project with Gradle
+
+Spine provides build-time tools for code generation and analysis. These tools are implemented
+as [Gradle plugins](https://docs.gradle.org/current/userguide/plugins.html) — an extendable, uniform
+way to tap into a project build process.
+
+The minimal Gradle configuration you will need to start a new project is:
+
+<embed-code file="examples/hello/build.gradle" start="plugins" end="}"></embed-code>
+```groovy
+plugins {
+    id 'io.spine.tools.gradle.bootstrap' version '1.5.24'
+}
+```
+
+Place the config into your root `build.gradle` file and execute a Gradle build. This will apply
+the Spine Bootstrap plugin to your project.
+
+You can also find this declaration on the [Gradle Plugin Portal](https://plugins.gradle.org/plugin/io.spine.tools.gradle.bootstrap),
+or on our [Getting Started page]({{site.baseurl}}/docs/quick-start).
+
+## Spine Bootstrap plugin 
+
+Spine Bootstrap plugin (Bootstrap for short) can configure a Spine Java server and JS client for
+you.
+
+The idea is to use Gradle subprojects for different parts of the system.
+
+### Model definitions
+
+One subproject would contain Protobuf definitions for the domain model. This subproject is
+typically called `model`. If the system contains more than one Bounded Context, there could be
+many subprojects, e.g. `model-users`, `model-trains`, etc. In `build.gradle` for those
+subprojects, declare:
+```groovy
+spine.assembleModel()
+```
+If one of the Bounded Contexts shares some domain language with another, add a dependency between
+them. This way, the downstream context may use Protobuf definitions of the upstream context.
+```groovy
+dependencies {
+    implemetation(project(':model-users'))
+}
+```
+
+### Java implementation
+
+Each Bounded Context should have a separate Gradle subproject for implementing server-side
+business logic. For example, `users` , `trains`, etc. In `build.gradle` for those
+subprojects, declare:
+```groovy
+spine.enableJava().server()
+
+dependencies {
+    protobuf(project(':model'))
+}
+```
+This will add dependencies to the `spine-server` artifact and set up code generation for
+the domain model types. Note that it is perfectly normal to have more Protobuf types in
+these modules, as long as those types are internal to the server and are not a part of
```

Comment:
This is a tricky part. As a first-time reader, would you want to learn about the existence of "internal" Proto types from this article? I think, not.